src/globus_cli/termio/server_timing.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import os
+import shutil
+
+import click
+import globus_sdk
+
+from .context import should_show_server_timing
+
+_BORDER_COLOR = "blue"
+_FILL_COLOR = "yellow"
+
+
+class ServerTimingParseError(ValueError):
+    pass
+
+
+def timing_string_to_dict(server_timing_string: str) -> dict[str, float]:
+    """
+    Given a Server Timing value as a string, parse it into a dict of the format
+      nice_name: value
+
+    For example
+
+      'a=1, "alpha"; b=2'
+
+    will parse as
+
+      {"alpha": 1, "b": 2}
```

Comment:
I'm confused by this example and the code below. It seems like the code below should split the string on semicolons so `"alpha"` is kept with `a=1`. However, it splits the overall string on comma, which would result in:

```python
[
  'a=1',
  '"alpha"; b=2',
]
```

I was anticipating this to be the result of line 51:

```python
[
  'a=1, "alpha"',
  'b=2',
]
```

Reply:
Oh. Huh. The code does not match this doc. Confusing indeed!

The Server-Timing header spec lets you use "nice names" which are quoted strings after the key-value pairs. (I have little idea why the spec is written this way, since it seems like the nice names aren't that important, but it doesn't bother me much.)

A good example might be
```
x0=50.2, "Authentication Callout"; x1=1.8, "Database Read"

=>

Authentication Callout: 50.2ms
Database Read: 1.8ms
```

I'm guessing that at one point I meant to use the nice names and then later I decided to ignore them.
But even if that's the output we want (and I think we should prefer the nice names, per the example above?), the parse should still start with semicolons.

Reply:
Er... I went to double-check the spec and it seems that I'm a bit wrong about what's allowed and supported.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing

:crossed_fingers: that our Server-Timing header builders in our applications are correct. I think they are -- I was looking much more at the spec when I wrote them.
But this all needs a bit of reassessment.

Reply:
I've started to dig deeper. It seems like the Server-Timing spec has changed in the time that I've been using it.
And it's continuing to change:
https://www.w3.org/standards/history/server-timing

Here's one of the older drafts which looks similar to what I remembered:
https://www.w3.org/TR/2017/WD-server-timing-20170725/

For right now, I'm writing down a method, `parse_metric_draft_2017` which declares itself as matching that draft.
I haven't figured out how we'll detect which draft is in use though, or if we will even want to do so.

The idea of writing a library with full detection and support for all of these drafts is simultaneously fun and not where I think we want to be spending our time. :sweat_smile:

Reply:
I just finished fairly thoroughly rewriting the parsing code. It should be a bit more readable (I hope) and makes it possible to support multiple spec versions in the future.

I'm going to resolve this because the relevant docstring was moved and rewritten, but the new parsing behavior would definitely benefit from a careful review. I did test it against live services and saw good results, however.